hack/generator/pkg/jsonast/jsonast.go

```diff
@@ -488,14 +488,7 @@ func generateDefinitionsFor(
 
 	// Give the type a name:
 	if isResource {
-		if specType, ok := result.(*astmodel.StructType); ok {
```

Comment:
I have a branch where I am introspecting resources based on their properties (and I have used this if statement to ensure that it's a struct and thus has properties).

Possibly that code needs to move -- what type(s) are the Kusto resources?

Reply:
I'm going to hold off on this because my simplification PR removes this check entirely.

Reply:
@matthchr take a look at #183 since this code/version is obsoleted